### PR TITLE
refactor: Remove `public` visibility of JUnit 5 tests

### PR DESCRIPTION
**Pull request by timtebeek**

### CODE REVIEW

**Comment by timtebeek** [comment]
I'm not as pleased with all the white space changes also brought in; especially as we [don't define an explicit style](https://github.com/moderneinc/jenkins-ingest/blob/main/repos.csv#L8193).
What might be causing these white space changes, and can they be reduced?
Tagging @kunli2 for his recent experience & insights there.

**Comment by timtebeek** [comment]
Compilation fails as `JavaTypeSignatureBuilderTest` and `JavaTypeMappingTest` should not have reduced visibility for Groovy. That's another issue we might need to workaround in the recipe.
```
/home/tim/Documents/workspace/rewrite/rewrite-groovy/src/test/java/org/openrewrite/groovy/GroovyTypeSignatureBuilderTest.java:24: error: JavaTypeSignatureBuilderTest is not public in org.openrewrite.java; cannot be accessed from outside package
import org.openrewrite.java.JavaTypeSignatureBuilderTest;
                           ^
/home/tim/Documents/workspace/rewrite/rewrite-groovy/src/test/java/org/openrewrite/groovy/GroovyTypeSignatureBuilderTest.java:35: error: JavaTypeSignatureBuilderTest is not public in org.openrewrite.java; cannot be accessed from outside package
public class GroovyTypeSignatureBuilderTest implements JavaTypeSignatureBuilderTest {
                                                       ^
/home/tim/Documents/workspace/rewrite/rewrite-groovy/src/test/java/org/openrewrite/groovy/GroovyTypeMappingTest.java:23: error: JavaTypeMappingTest is not public in org.openrewrite.java; cannot be accessed from outside package
import org.openrewrite.java.JavaTypeMappingTest;
                           ^
/home/tim/Documents/workspace/rewrite/rewrite-groovy/src/test/java/org/openrewrite/groovy/GroovyTypeMappingTest.java:34: error: JavaTypeMappingTest is not public in org.openrewrite.java; cannot be accessed from outside package
class GroovyTypeMappingTest implements JavaTypeMappingTest {
                                       ^
warning: unknown enum constant When.MAYBE
  reason: class file for javax.annotation.meta.When not found
4 errors
4 warnings
```

File: rewrite-java-test/src/test/java/org/openrewrite/java/recipes/FindRecipesTest.java [hunk 1]
```diff
@@ -23,25 +23,25 @@
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.openrewrite.java.Assertions.java;
 
-public class FindRecipesTest implements RewriteTest {
+class FindRecipesTest implements RewriteTest {
```
Comment (timtebeek) [inline rewrite-java-test/src/test/java/org/openrewrite/java/recipes/FindRecipesTest.java:26]:
```suggestion
public class FindRecipesTest implements RewriteTest {
```

**Comment by shanman190** [comment]
@timtebeek, I believe that the whitespace issues are as a result of the continuation indent calculation by Auto detect. Since rewrite tests use a 2 space continuation indent, it's probably that a majority of files are using the default 8 space indent, so then auto format is trying to align the style to the intellij default.

**Comment by kunli2** [comment]
I think the 2-space continuation indent is set [here](https://github.com/openrewrite/rewrite/blob/main/rewrite-java-test/src/test/java/org/openrewrite/java/.editorconfig#L5). and this overrides the default Intellij default java code style.

**Comment by shanman190** [comment]
@kunli, that is correct, but editorconfig files only apply for intellij itself presently. To be explicit there isn't an editorconfig NamedStyle. So as a result during parsing auto detect tries to determine the formatting which is just a heuristic match based on "popularity" in files within a given language and source set. Then uses that for the formatting in AutoFormat. I'm suspecting that there are many files still in the test source set for each module that still are using the old 8 space continuation. Not totally for sure though since I'm not in front of my PC.

**Comment by kunli2** [comment]
@shanman190, An. I see your point, I thought this PR is made by manually changing via Intellij, to remove `public` and do IDE auto-format in the meantime, but not from OpenRewrite `AutoFormat` recipe.

**Comment by shanman190** [comment]
@kunli2, exactly. In Tim's initial commit it was done via the OpenRewrite recipe `org.openrewrite.java.testing.cleanup.TestsShouldNotBePublic` which also triggers a OpenRewrite `AutoFormat` run which is the culprit of the undesired continuation indent modification.

**Comment by timtebeek** [comment]
Thanks for the hints @shanman190 ! So what I take from this is that we occasionally have .editorconfig files in for instance [rewrite-core/src/test/java/org/openrewrite/.editorconfig](https://github.com/openrewrite/rewrite/blob/main/rewrite-core/src/test/java/org/openrewrite/.editorconfig), which are not taken into account by the auto determined & applied formatter. This leads to conflicts between what IntelliJ applied previously for those files, as configured, and what the recipe changes want to apply now.

There's then now a couple different ways forward:
1. we consistently apply the .editorconfig settings throughout, to ensure the auto formatter picks that as the dominant form
2. we remove the .editorconfig to consistently apply the defaults throughout
3. we alter the style detection to take .editorconfig into account
4. we alter the style detection to look per source folder, rather than per project, to implicitly pick up .editorconfig styles
5. ... possibly more options ... 
Not sure which of these would be preferred; any input welcome! 

As is it seems we can't simply apply these changes without conflicts either in IDE or in a subsequent recipe run.

**Comment by shanman190** [comment]
So formatting is discovered per project/sourceSet/language tuple. So it's probably just that there are _enough_ files that exist in that project sourceSet that as of yet that still have the old formatting. 

The easiest of solutions is to format the rewrite project to a desired style, then `AutoFormat` should then maintain that style based upon the aforementioned qualities. An alternative would be to use an explicit style, BUT this has an issue that the OpenRewrite project uses two styles based upon which source set you are in presently.

**Comment by sambsnyd** [comment]
I took a pass through rewrite-testing-frameworks and removed several instances of overly aggressive reformatting. Give it another try and see if that helps
https://github.com/openrewrite/rewrite-testing-frameworks/commit/898f8639e6ded5dc3f757846d7b3268f72712ec3

**Comment by timtebeek** [comment]
Thanks for that @sambsnyd ! [Tried it again](https://public.moderne.io/results/Iijp1); still some unfortunate changes that don't pick up on the .editorconfig setting of `ij_continuation_indent_size = 2`, maybe because the project isn't consistent across all source sets, but I don't know the details on what exactly to expect there.

**Comment by timtebeek** [comment]
Going to close this one due to conflicts. Should be easy to recreate from SaaS, hopefully without some of the whitespace issues seen before.

**Pull request closed without merging**